adcc/backends/pyscf.py

```diff
@@ -262,7 +262,7 @@ def import_scf(scfres):
 
 
 def run_hf(xyz, basis, charge=0, multiplicity=1, conv_tol=1e-11,
-           conv_tol_grad=1e-9, max_iter=150, pe_options=None):
+           conv_tol_grad=1e-9, max_iter=150, pe_options=None, pcm=None):
```

Comment:
I think we need some more general option passing for the environment here...
Maybe an `environment` dictionary with keywords `model` and `options` would be better.
The function is supposed to be `adcc`-internal, so we can decide what we want. If you have 2mins, can you give this a quick look, @mfherbst?

Reply:
Yes I agree. It's running out of hands with the arguments. Perhaps an `environment` dict with a key `pe` and `pcm` to the respective options ?

Reply:
I thought about it again... Maybe generalizing this does not make sense, because the handling of PCM options is different in each host program. The reason why we can do this for PE is that CPPE is used throughout. So, for now I suggest to completely remove PCM stuff from the `run_hf` and `cached_backend_hf` functions, and implement some routine that runs the PCM-HF in the test file.
Anything else would be overengineering for things we don't even support right now.

Reply:
Do you think we need a lot of different PCM calculations for the tests?
If yes, I agree with you @maxscheurer. Otherwise I think it would be better to just add the few extra lines in the corresponding `run_hf` and then maybe use the dict with a bool or sth like that to trigger PCM

Reply:
Misunderstood the comments... yes you are right @maxscheurer, a dict of this form would not make sense for PCM.
But at least at the moment with only PCM and PE it would be possible to use an if for PCM and else for PE. Depends on the other envs that you plan to add if this makes sense. And of course the PCM options would be rather fixed in `run_hf`, though maybe 1 or 2 options could be passed with the dict... not sure right now.

Reply:
I see your point, but I don't like that we would need to hard-code PCM parameters in the respective backends, even if it's just for testing purposes. The reason is that it could be 'misused' as a wrapper for running HF with some backend, and would produce unpredictable results in that case. Another point is that you'd need to raise `NotImplementedError`s for the unsupported backends etc., which would result in quite some code addition for just 1-2 test cases.
So, I'd say make a simple PCM-HF routine in your test file which serves the purpose - in my opinion one can generalize it when the time comes 😉

Reply:
ok good points.